### PR TITLE
Add documentation to Charmhub

### PR DESCRIPTION
**Pull request by gatici**
# Description

This change adds the basic information to Charmhub related with the Self Signed X.509 Certificate Operator.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,16 +15,62 @@ This charm is useful when developing charms or when deploying charms in non-prod
...
+
+```shell
+juju remove-application self-signed-certificates
+```
+
+If required, you can remove the deployment completely by running:
+
+```shell
+juju destroy-model certificates
+```
```
Comment (gruyaume) [inline README.md:41]: I don't think this is necessary as it is the case for all charms and not specific to this one. I'd keep the doc specific to the charm here.

File: docs/explanation/e-interfaces.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Interfaces/Endpoints
```
Comment (gruyaume) [inline docs/explanation/e-interfaces.md:1]: Similar comment to the `config` one, there is an `integrations` tab in charmhub that is automatically generated.

File: docs/how-to/h-manage-units.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+To deploy a single unit of Self Signed Certificates Operator using its default configuration.
+```shell
+juju deploy self-signed-certificates --channel edge
+```
+
+## Replication
+
+Both scaling-up and scaling-down operations are performed using `juju scale-application`:
+```shell
+juju scale-application  self-signed-certificates <desired_num_of_units>
```
Comment (gruyaume) [inline docs/how-to/h-manage-units.md:14]: Are we sure this actually works?

File: docs/how-to/h-manage-app.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Manage Self Signed Certificates Operator
```
Comment (gruyaume) [inline docs/how-to/h-manage-app.md:1]: I'm not sure of the value of this file

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+The Self Signed Certificates Operator provides self-signed X.509 certificates to your charms.
```
Comment (gruyaume) [inline docs/index.md:1]: Ultimately I think most of our docs for this simple charm can be in this index. It's a fairly simple charm at the end of the day and we probably don't want to maintain 5-10 pages of text for it.

File: docs/tutorial/t-deploy.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Make Self Signed Certificates Operator up and running
```
Comment (gruyaume) [inline docs/tutorial/t-deploy.md:1]: As mentioned in other places, what is here could be in the index. We probably don't want a full page to tell people how to deploy charms.

**Review by gruyaume — CHANGES_REQUESTED** [review]
Good job, this is very thorough.

However I think there is too much text. This is a fairly simple charm at the end of the day and we probably don't want 5-10 pages of text for it. 

All the documentation that we add has to be maintained over time and the more concise we are, the less effort we'll have to place down the line.

We should also leverage all the automation we can. For instance, config and integrations are automatically generated in Charmhub. We shouldn't need to write our own docs about these subjects in addition to those.

In general I believe all of our documentation for this charm could hold on 1 page.

File: docs/how-to/h-manage-units.md (hunk 3 above)
Comment (gatici) [reply docs/how-to/h-manage-units.md:14]: In line 17, I commented that it only works with 1 unit. This command is also used to scale down so it's reasonable to add.

File: README.md (hunk 1 above)
Comment (gatici) [reply README.md:41]: OK, I will rollback to the previous version.

File: docs/explanation/e-configure.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Set the Self Signed Certificates Operator configuration options
```
[the remark this answers is not in the document]
Comment (gatici) [reply docs/explanation/e-configure.md:1]: OK, I will remove it.

File: docs/explanation/e-interfaces.md (hunk 2 above)
Comment (gatici) [reply docs/explanation/e-interfaces.md:1]: I will remove it.

File: docs/how-to/h-manage-units.md (hunk 3 above)
Comment (gatici) [reply docs/how-to/h-manage-units.md:14]: I will write it more explicitly that scale up is  not supported yet.

File: README.md (hunk 1 above)
Comment (gatici) [reply README.md]: Done

File: docs/explanation/e-configure.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (gatici) [reply docs/explanation/e-configure.md]: Done

File: docs/explanation/e-interfaces.md (hunk 2 above)
Comment (gatici) [reply docs/explanation/e-interfaces.md]: Done

File: docs/how-to/h-manage-units.md (hunk 3 above)
Comment (gatici) [reply docs/how-to/h-manage-units.md]: Done. The content is moved to tutorial overview section.

File: docs/how-to/h-manage-app.md (hunk 4 above)
Comment (gatici) [reply docs/how-to/h-manage-app.md]: Removed.

**Comment by gatici** [comment]
> Good job, this is very thorough.
> 
> However I think there is too much text. This is a fairly simple charm at the end of the day and we probably don't want 5-10 pages of text for it.
> 
> All the documentation that we add has to be maintained over time and the more concise we are, the less effort we'll have to place down the line.
> 
> We should also leverage all the automation we can. For instance, config and integrations are automatically generated in Charmhub. We shouldn't need to write our own docs about these subjects in addition to those.
> 
> In general I believe all of our documentation for this charm could hold on 1 page.

Review comments are addressed in the commit  https://github.com/canonical/self-signed-certificates-operator/pull/23/commits/21b5179b52cf813b8babfce13557d88d0703cbc3. Ready to review again.

File: docs/tutorial/t-get-signed-certificates.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Get Signed Certificates
```
Comment (gruyaume) [inline docs/tutorial/t-get-signed-certificates.md:1]: I'm not sure this section is useful. There's an actions tab in Charmhub (https://charmhub.io/self-signed-certificates/actions?channel=edge)

File: docs/tutorial/t-setup-environment.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
+# Environment Setup
+
+This is part of the [Self Signed X.509 Certificates Operator Tutorial](/t/self-signed-x-509-certificates-tutorial-overview/11600?channel=edge). Please refer to this page for more information and the overview of the content.
+
+## Minimum requirements
+Before we start, make sure your machine meets the following requirements:
+- Ubuntu 22.04 (Jammy) or later.
+- 8GB of RAM.
+- 2 CPU threads.
+- At least 20GB of available storage.
```
Comment (gruyaume) [inline docs/tutorial/t-setup-environment.md:10]: 20GB of storage!!?

File: docs/tutorial/t-setup-environment.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
+# Environment Setup
```
Comment (gruyaume) [inline docs/tutorial/t-setup-environment.md:1]: I'm not sure of the value of having a tutorial for this charm. IMO a simply Usage section would be useful. Here we mostly have a tutorial on how to get started with Juju and that's probably not what we want to have on every single charm on Charmhub.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: docs/tutorial/t-get-signed-certificates.md (hunk 8 above)
Comment (gatici) [reply docs/tutorial/t-get-signed-certificates.md]: The section is removed.

File: docs/tutorial/t-setup-environment.md (hunk 9 above)
Comment (gatici) [reply docs/tutorial/t-setup-environment.md]: This section is removed.

File: docs/tutorial/t-setup-environment.md (hunk 10 above)
Comment (gatici) [reply docs/tutorial/t-setup-environment.md]: t-setup-environment.md file is removed.

File: docs/tutorial/t-overview.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+## How to manage units
+
+If Self Signed Certificates Operator is deployed with its default configuration, single unit exists.
+The Self Signed Certificates Operator works with single replica for the moment and scale up operation is not supported yet.
+
+Scaling-down operation is performed using following command:
+```shell
+juju scale-application  self-signed-certificates 0
+```
```
Comment (gruyaume) [inline docs/tutorial/t-overview.md:50]: This block is unnecessary, it's applicable to all charms

File: docs/tutorial/t-overview.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+## Step-by-step guide
+
+Here’s an overview of the steps required with links to our separate tutorials that deal with each individual step:
+* [Deploy Self Signed Certificates Operator](/t/self-signed-x-509-certificates-tutorial-overview/11600?channel=edge)
+* [Configure Self Signed Certificates Operator](https://charmhub.io/self-signed-certificates/configure?channel=edge)
+* [Managing your units](/t/self-signed-x-509-certificates-tutorial-overview/11600?channel=edge)
+* [Relate Self Signed Certificates Operator with other applications](https://charmhub.io/self-signed-certificates/integrations?channel=edge)
+* [Get Signed Certificates](https://charmhub.io/self-signed-certificates/actions?channel=edge)
```
Comment (gruyaume) [inline docs/tutorial/t-overview.md:80]: This is not needed anymore

File: docs/index.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+This charm relies on the `tls-certificates` charm relation interface. When a requirer charm inserts a Certificate Signing Request in its unit databag, the `self-signed-certificates-operator` will read it, generate a self-signed X.509 certificates and
+inserts this certificate back into the relation data.
+
+This charm [Self-Signed-Certificates-Operator](https://github.com/canonical/self-signed-certificates-operator) is useful when developing charms or when deploying charms in non-production environment on top of [Juju](https://juju.is/).
```
Comment (gruyaume) [inline docs/index.md:6]:
```suggestion
The [Self-Signed-Certificates-Operator](https://github.com/canonical/self-signed-certificates-operator) is useful when developing charms or when deploying charms in non-production environment on top of [Juju](https://juju.is/).
```

File: docs/tutorial/t-overview.md
Comment (gruyaume) [inline docs/tutorial/t-overview.md:1]: Right now this file is not really a tutorial. A tutorial would walk you through getting started using the charm, here we have a mix of explanations and how to use various Juju commands and we expect the user to already have charm they want relate self-signed-certificates to (`<your charm which needs tls certificates>`).

Overall why not place everything that is necessary in a short "Usage" section in the index?

All we need is two lines:

```shell
juju deploy self-signed-certificates --channel edge
juju relate self-signed-certificates <your charm which needs tls certificates>
```

File: docs/how-to/.gitkeep
Comment (gruyaume) [inline docs/how-to/.gitkeep:1]: Do we need to keep the how-to, explanations directories if there's nothing in them?

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: docs/tutorial/t-overview.md (hunk 11 above)
Comment (gatici) [reply docs/tutorial/t-overview.md:50]: Here, we are indicating that Self Signed Certificates Operator does not support scaling up which is not common. (line 45)

File: docs/how-to/.gitkeep
Comment (gatici) [reply docs/how-to/.gitkeep:1]: I was trying to protect diataxis structure. For that charm, it looks like we do not need some sections for the moment, but I do not have strong opinion about removing them. I asked this question in the documentation channel.

File: docs/index.md (hunk 13 above)
Comment (gatici) [reply docs/index.md:6]: OK

File: docs/tutorial/t-overview.md (hunk 12 above)
Comment (gatici) [reply docs/tutorial/t-overview.md:80]: OK

File: docs/tutorial/t-overview.md (hunk 11 above)
Comment (gatici) [reply docs/tutorial/t-overview.md]: I just moved line 45 under Usage section in index.md.

File: docs/how-to/.gitkeep
Comment (gatici) [reply docs/how-to/.gitkeep:1]: After discussing in the documentation channel: https://chat.canonical.com/canonical/pl/5nue9xwuxbrqdcubgfrxo33nfr,  all empty sections are removed.
https://docs.google.com/document/d/1V8oMjgecQdCxdilPmu78vCW2wvG_Kc13qlxeNRVx490/edit#heading=h.dfrhe8pys6o4

**Review by gruyaume — APPROVED** [review]

**Pull request merged**